### PR TITLE
fix: redo assertHostnames in HostnameMergeSuite.TestMerge

### PR DESCRIPTION
**Pull request by DmitriyMV**
Use `rtestutils.AssertResources` for hostnames test.

Signed-off-by: Dmitriy Matrenichev <dmitry.matrenichev@siderolabs.com>

### CODE REVIEW

**Comment by DmitriyMV** [comment]
I updated as many tests as I could, but there is just too many of them for one go.

File: internal/app/machined/pkg/controllers/network/rtestutils/assert.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+// AssertResources asserts on a resource list.
+func AssertResources[R ResourceWithRD](
+	ctx context.Context,
+	t *testing.T,
+	st state.State,
+	ids []resource.ID,
+	assertionFunc func(r R, assertion *assert.Assertions),
+	opts ...Option,
```
Comment (DmitriyMV) [inline internal/app/machined/pkg/controllers/network/rtestutils/assert.go:38]: I'm not sure if we have any other places that require us to pass Namespace separately, and those I don't know if it make sense to move it into runtime project.

Comment (smira) [reply internal/app/machined/pkg/controllers/network/rtestutils/assert.go:38]: I think certainly we shouldn't have a copy here

Comment (DmitriyMV) [reply internal/app/machined/pkg/controllers/network/rtestutils/assert.go:38]: Okay - moved to https://github.com/cosi-project/runtime/pull/236

**Pull request merged**